server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -592,10 +592,9 @@ private SnapshotInfo inProgressSnapshot(SnapshotsInProgress.Entry entry) {
      * @return map of shard id to snapshot status
      */
     public Map<ShardId, IndexShardSnapshotStatus> snapshotShards(final String repositoryName,
+                                                                 final RepositoryData repositoryData,
```

Comment:
Note: this is a small optimization in the case of the Snapshot Status API where the repository index was loaded two times.